advanced-topics/network-configuration.md

```diff
@@ -0,0 +1,145 @@
+
+Frequently, VPC peering APIs require all subnets to be non-overlapping, which isn't guaranteed from our defaults.  This can always be updated in `bootstrap/terrraform/main.tf` (As a reference, CIDR blocks follow this format: `10.xx.xx.xx/yy`, the examples below are placeholders.)
+
+{% tabs %}
+{% tab title="AWS" %}
+On AWS, update the `aws-bootstrap` modules configuration with:
+
+```bash
+public_subnets = ["your.cidr.pub.1", "your.cidr.pub.2", "your.cidr.pub.3"]
+private_subnets = ["your.cidr.priv.1", "your.cidr.priv.2", "your.cidr.priv.3"]
+worker_private_subnets = ["your.cidr.worker.1", "your.cidr.worker.2", "your.cidr.worker.3"]
+```
+{% endtab %}
+
+{% tab title="GCP" %}
+On GCP, update the `gcp-bootstrap` modules configuration with:
+
+```bash
+vpc_subnetwork_cidr_range = "your.cidr"
+
+# you might also want to update cluster_secondary_range_cidr 
+# and services_secondary_range_cidr
+```
+
+{% endtab %}
+
+{% tab title="Azure" %}
+With Azure, update the `azure-bootstrap` modules configuration with:
+
+```bash
+address_space = "your.cidr"
+subnet_prefixes = ["your.cidr.pref"]
+```
+
+{% endtab %}
+{% endtabs %}
+
+{% hint style="warning" %}
+💡 Note that updating these will likely cause the VPC to be replaced, which will recreate your cluster. We recommend that you destroy your cluster before applying network config modifications to it.
+{% endhint %}
+
+## Configuring VPC Peering
+
+Plural creates a fresh VPC and Kubernetes cluster for deploying your applications into. This ensures that we have a clean environment to deploy into and minimizes disruption to existing systems. It does come with the tradeoff of slightly increased network complexity, but most cloud providers can bridge this using VPC peering. This is a technology that effectively allows you to combine the address spaces of two VPCs in the cloud you operate in.
+
+**There is a caveat:** the VPCs should have no overlapping subnets in addition to some other complexities per cloud provider. Refer to the guide above on subnet configuration before proceeding.
+
+Plural makes it easy to add additional Terraform to the stacks we generate.  Effectively, as long as you don’t modify the `[main.tf](http://main.tf)` file at the root of a Terraform folder, or any of the module folders, we’ll preserve your Terraform between builds and apply it for you appropriately as changes arise.  For configuring a VPC peer, let’s create a new Terraform file called `[network.tf](http://network.tf)` and drop the configuration in there for your respective cloud:
+
+{% tabs %}
+{% tab title="AWS" %}
+```bash
+resource "aws_vpc_peering_connection" "foo" {
+   peer_owner_id = "my-owner-id" # Use appropriate values here.
+   peer_vpc_id   = "peer-vpc-id"
+   vpc_id        = module.aws-bootstrap.vpc.id
+}
+```
+{% endtab %}
+
+{% tab title="GCP" %}
+```bash
```

Comment:
I think we might be able to use terraform as a codeblock in gitbook (notion didn't support it)

Reply:
I checked, it wasn't available unfortunately.

Reply:
is hcl available?

Reply:
I think it is, we should use that, it's basically the same